compiler/rustc_middle/src/ty/generic_args.rs

```diff
@@ -281,7 +298,8 @@ impl<'tcx> GenericArg<'tcx> {
     pub fn is_non_region_infer(self) -> bool {
         match self.unpack() {
             GenericArgKind::Lifetime(_) => false,
-            GenericArgKind::Type(ty) => ty.is_ty_or_numeric_infer(),
+            // FIXME: This shouldn't return numerical/float.
```

Comment:
it should or it shouldn't? The fixme is either out of date or wrong. I think it's out of date